### PR TITLE
Connection Unit test

### LINKED ISSUE #8147: [Unit test] Adding more Unit tests for handlers package  in the server Component
#### Current Behavior
We have few unit tests written in the [handlers package in the server component](https://github.com/meshery/meshery/tree/master/server/handlers)
#### Desired Behavior
Would propose adding more unit tests to help improve our ci builds 

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
- Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)
- 🧪 [Meshery Test Plan Spreadsheet](https://docs.google.com/spreadsheets/d/13Ir4gfaKoAX9r8qYjAFFl_U9ntke4X5ndREY1T7bnVs/edit#gid=0)

### PR DESCRIPTION
**Pull request by Philip-21**
**Notes for Reviewers**

This is PR fixes for #8147 
This is the unit test for the [connection_handlers.go](https://github.com/meshery/meshery/blob/master/server/handlers/connections_handlers.go) in the server component

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by Philip-21** [comment]
I just want to test  the `saveConnection()` before testing the other functions

File: server/handlers/connections_handlers_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	Type:    "type-value",
+	MetaData: map[string]interface{}{
+		"key1": "value1",
+		"key2": 123,
+	},
+	Status: models.CONNECTED,
+	CredentialSecret: map[string]interface{}{
+		"secretKey1": "secretValue1",
+		"secretKey2": true,
+	},
```
Comment (theBeginner86) [inline server/handlers/connections_handlers_test.go:27]:
```suggestion
	CredentialSecret: map[string]interface{}{
		"token": "eyJhY2Nlc3NfdG9rZW4iOiJleUpoYkdjaU9pSlNVekkxTmlJc0ltdHBaQ0k2SW5CMVlteHBZenBvZVdSeVlTNXFkM1F1WVdOalpYTnpMWFJ2YTJWdUlpd2lkSGx3SWpvaVNsZFVJbjAuZXlKaGRXUWlPbHRkTENKamJHbGxiblJmYVdRaU9pSnRaWE5vWlhKNUxXTnNiM1ZrSWl3aVpYaHdJam94TmpnNU56TXlNRE16TENKbGVIUWlPbnQ5TENKcFlYUWlPakUyT0RrMk5EVTJNeklzSW1semN5STZJbWgwZEhBNkx5OXNiMk5oYkdodmMzUTZPVEF3TUM4aUxDSnFkR2tpT2lJek9UVmpOMlV3TnkwME1XTXhMVFF6T1RrdE9ERXhaQzFqWmpKak56ZGlabVZtTVRraUxDSnVZbVlpT2pFMk9EazJORFUyTXpJc0luTmpjQ0k2V3lKdmNHVnVhV1FpTENKdlptWnNhVzVsSWl3aWIyWm1iR2x1WlY5aFkyTmxjM01pWFN3aWMzVmlJam9pWWxkV2VtRkhNV2hqUlVKeldWaHNiR05xVlhWaFZ6ZzlPbUpYVm5waFIxWjVaVk14YW1KSE9URmFRVDA5SW4wLmpVelhmMkRrTi15NVpiWThSVkFuTjJZeXFTS05BT2tCZ0dDNHMxb2dnUFoxQS1CakdNbkFyMTZjRVBLcmVPVENXSzNfaU1DdWctcmltZG9QeGlqWFllX1d5bGpqeU9Ta1RyR0xpY0xubE1WNmhXNndJbXptT0E1VWh2UDVWZ09oeXRDZ1VQU3U0OFRVRFFTUE1NU1VkRnRFTzY2c0tvM3pXeURiV01lbm1rTHp6bGJwaFhFbXJpTDFhcnZkRHBFZV9ndHlWa0xjbktCS3p6U2Q2WWF3eDUzS0NhWGhGa2Q0Y0FVaUFVX2lGZjZnYW9ib0VRNUVfV2ZtRDJJaG9BaEtTWHJNbjdmTkxJTERwOFVXRUk3VGdwUUVFZ1QyY1VXdWY4eXlhV2NxOWpmRW9ia3ZQLXpTa29oXzJYZWlLRmtFRjAydTJ2VGp6Z3VpSmJRRl9DWmtQZGlVcGt1VlU0VEJHZVk5U2h6SWxUeGxYdWNySnp1UDI0YVJOMjl6cnNxN1hUMTktTVd2WVRkb3RCek1Dd3hhOTV2Q1BkdUFOTXhPTzVoYWJ3X2hfVUoyTGwycFFfT1dkeHJUSjFNV2pQR25hdzBFcGtqcVlDV1lSdzVJMnBwZGctWGRJUzM3aEpBbTk0MXpLNFhVanhIZzluTVJxSE53Y1RjWnFYRGU0UUstejBVWTJ0Nm0tNjBTZjZJRDg1anpkeFlxMmVfQ3BFMno3MFhaSmJZT1A1OGthU0Zfekxac1FUSkwyUFBDRmpGOGl2bnFFRzlzLUZqemRSSUxmUi1Tc2VsUjZzWk4tb3JteXlLbWF4MnlIYmljZkxTRjhveWQzWW5QSHk5ZWpYSkY1bVpwd3lPWUpmS1VRNEtOVTZ3QXg2cjljRnVvQ0NSRjJFRnJDdnJTVWE0IiwidG9rZW5fdHlwZSI6ImJlYXJlciIsInJlZnJlc2hfdG9rZW1iOiIteXI0Z2VDeEExWlJPTFJyeTlsa3doUXl5VGR4SzV4clFzNzY1ejBkYUxnLmEzSVdLVGZCVHQ5VzRkZ0Mwdl93U0dZX1ozYTZPSndWYl9jU1h4c1hUalUiLCJleHBpcnkiOiIyMDIzLTA3LTE5VDA3OjMwOjMyLjkxNjozNpswNToqMCA1",
	},
```

Comment (theBeginner86) [reply server/handlers/connections_handlers_test.go:27]: Though this is not real token but just to mimic real data, I have added something which is closest to the real deal

**Review by theBeginner86 — COMMENTED** [review]
I have switched dummy data with real one 👍

File: server/handlers/connections_handlers_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/layer5io/meshery/server/models"
+)
+
+var testcon = &models.ConnectionPayload{
+	Kind:    "meshery",
+	SubType: "management",
+	Type:    "platform",
+	MetaData: map[string]interface{}{
+		"key1": "value1",
+		"key2": 123,
```
Comment (theBeginner86) [inline server/handlers/connections_handlers_test.go:21]:
```suggestion
	MetaData: map[string]interface{}{
    "server_id": "d058accd-290d-4123-9c19-3ee7c0b1cbdb",
    "server_version": "v0.6.91",
    "server_location": "https://playground.meshery.io",
    "server_build_sha": "4bf9a717"
```

**Review by theBeginner86 — APPROVED** [review]
Lgtm

**Comment by theBeginner86** [comment]
Thank you @Philip-21 🙌

**Comment by Philip-21** [comment]
Thank you for the reviews @theBeginner86

**Comment by Philip-21** [comment]
@theBeginner86 it seems everything is fine in the code cov report,  tests are passed and confirmed

**Comment by vishalvivekm** [comment]
@Philip-21 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by Philip-21** [comment]
> @Philip-21 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

Alright

**Comment by theBeginner86** [comment]
> @theBeginner86 it seems everything is fine in the code cov report, tests are passed and confirmed

@Philip-21 
I'm seeing the nil ptr error for our new SaveConnection unit test. https://github.com/meshery/meshery/actions/runs/5585719522/jobs/10208838470?pr=8196#step:5:43
Would you please check it?

> coming from line 46 in server/handlers/connections_handlers_test.go

**Comment by Philip-21** [comment]
@theBeginner86 from my take, i feel it is a mesheryctl issue , or its either that same one i mentioned earlier. 
Looking at this error it seems that its from the `provider.saveconnection()`. 
`github.com/layer5io/meshery/server/handlers.(*Handler).SaveConnection(0xc000953e28, {0x57f63c0, 0xc000ab6f80}, 0xc00024ee00, 0x518152b?, 0x1d?, {0x0, 0x0})
	/home/runner/work/meshery/meshery/server/handlers/connections_handlers.go:39 +0x16d`

**Comment by suhail34** [comment]
As this is a server side testing also create an http mock server

**Review by suhail34 — CHANGES_REQUESTED** [review]

**Comment by Philip-21** [comment]
@theBeginner86 with these new additions I just made can you help add some dummy data to the  fields for  the user variable, which I will use as a parameter when  testing the `Testsaveconnection()` ,

File: server/handlers/connections_handlers_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	
+	"net/http"
+	"net/http/httptest"
+
+	"testing"
+
+	"github.com/layer5io/meshery/server/models"
+)
+
+var testcon = &models.ConnectionPayload{
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply server/handlers/connections_handlers_test.go]: @Philip-21 addressed?

**Pull request closed without merging**